internal/cephes/incbeta.go

```diff
@@ -0,0 +1,311 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+/*
+ * Cephes Math Library, Release 2.3:  March, 1995
+ * Copyright 1984, 1995 by Stephen L. Moshier
+ */
+
+package cephes
+
+import (
+	"math"
+
+	"github.com/gonum/mathext/internal/gonum"
+)
+
+const (
+	maxGam = 171.624376956302725
+	big    = 4.503599627370496e15
+	biginv = 2.22044604925031308085e-16
+)
+
+// Incbet computes the incomplete beta function.
+func Incbet(aa, bb, xx float64) float64 {
+	if aa <= 0 || bb <= 0 {
+		panic(badParamOutOfBounds)
+	}
+	if xx <= 0 || xx >= 1 {
+		if xx == 0 {
+			return 0
+		}
+		if xx == 1 {
+			return 1
+		}
+		panic(badParamOutOfBounds)
+	}
+
+	var flag int
+	if bb*xx <= 1 && xx <= 0.95 {
+		t := pseries(aa, bb, xx)
+		return transformT(t, flag)
+	}
+
+	w := 1 - xx
+
+	// Reverse a and b if x is greater than the mean.
+	var a, b, xc, x float64
+	if xx > aa/(aa+bb) {
+		flag = 1
+		a = bb
+		b = aa
+		xc = xx
+		x = w
+	} else {
+		a = aa
+		b = bb
+		xc = w
+		x = xx
+	}
+
+	if flag == 1 && (b*x) <= 1.0 && x <= 0.95 {
+		t := pseries(a, b, x)
+		return transformT(t, flag)
+	}
+
+	// Choose expansion for better convergence.
+	y := x*(a+b-2.0) - (a - 1.0)
+	if y < 0.0 {
+		w = incbcf(a, b, x)
+	} else {
+		w = incbd(a, b, x) / xc
+	}
+
+	// Multiply w by the factor
+	//  a      b   _             _     _
+	//  x  (1-x)   | (a+b) / ( a | (a) | (b) ) .   */
+	var t float64
+	y = a * math.Log(x)
+	t = b * math.Log(xc)
+	if (a+b) < maxGam && math.Abs(y) < maxLog && math.Abs(t) < maxLog {
+		t = math.Pow(xc, b)
+		t *= math.Pow(x, a)
+		t /= a
+		t *= w
+		t *= 1.0 / gonum.Beta(a, b)
+		return transformT(t, flag)
+	}
+
+	// Resort to logarithms.
+	y += t - gonum.Lbeta(a, b)
+	y += math.Log(w / a)
+	if y < minLog {
+		t = 0.0
+	} else {
+		t = math.Exp(y)
+	}
+
+	return transformT(t, flag)
+}
+
+func transformT(t float64, flag int) float64 {
+	if flag == 1 {
+		if t <= machEp {
+			t = 1.0 - machEp
+		} else {
+			t = 1.0 - t
+		}
+	}
+	return t
+}
+
+// incbcf is the continued fraction expansion #1 for incomplete beta integral
```

Comment:
I changed the comment. I think that's accurate, because it looks like the regularization comes later, but I'm not sure.